src/main/java/org/spongepowered/mod/text/selector/SpongeSelector.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.mod.text.selector;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import net.minecraft.command.PlayerSelector;
+import net.minecraft.server.MinecraftServer;
+
+import org.spongepowered.api.entity.Entity;
+import org.spongepowered.api.text.selector.Selector;
+import org.spongepowered.api.text.selector.SelectorBuilder;
+import org.spongepowered.api.text.selector.SelectorType;
+import org.spongepowered.api.util.annotation.NonnullByDefault;
+import org.spongepowered.api.world.Location;
+
+import com.google.common.base.Preconditions;
+
+@NonnullByDefault
+public class SpongeSelector implements Selector {
+
+    private final SelectorType type;
+    private final Map<String, String> arguments;
+    private final boolean requiresLocation;
+
+    public SpongeSelector(SelectorType type, Map<String, String> arguments, boolean requiresLocation) {
+        this.type = type;
+        this.arguments = arguments;
+        this.requiresLocation = requiresLocation;
+    }
+
+    @Override
+    public SelectorType getType() {
+        return this.type;
+    }
+
+    @Override
+    public Map<String, String> getArguments() {
+        return this.arguments;
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public List<Entity> resolve(Location location) throws IllegalArgumentException {
+        Preconditions.checkArgument(this.requiresLocation == true && location == null);
```

Comment:
Should be `!this.requiresLocation || location != null`.

Also, Preconditions should be statically imported.